sockeye/train.py

```diff
@@ -568,15 +604,18 @@ def check_encoder_decoder_args(args) -> None:
 
 
 def create_model_config(args: argparse.Namespace,
-                        vocab_source_size: int, vocab_target_size: int,
-                        config_data: data_io.DataConfig) -> model.ModelConfig:
+                        vocab_source_size: int,
+                        vocab_target_size: int,
+                        config_data: data_io.DataConfig,
+                        source_factor_vocab_sizes: Optional[List[int]] = []) -> model.ModelConfig:
     """
     Create a ModelConfig from the argument given in the command line.
 
     :param args: Arguments as returned by argparse.
-    :param vocab_source_size: The size of the source vocabulary.
+    :param vocab_source_size: The sizes of the source vocabularies.
```

Comment:
right now its still just one, but as said above, I would like it to be a list  to avoid the extra argument.
This probably makes a major version bump but this is a pretty big change anyway.